java/Lexer.java

```diff
@@ -17,55 +20,48 @@ public Lexer(String input) {
     public Token nextToken() {
         this.skipWhitespace();
 
-        Token t = null;
-        TokenType tt = TokenType.ILLEGAL;
-        switch (this.getCc()) {
-            case '{' -> tt = TokenType.LSQIRLY;
-            case '}' -> tt = TokenType.RSQIRLY;
-            case '(' -> tt = TokenType.LPAREN;
-            case ')' -> tt = TokenType.RPAREN;
-            case ',' -> tt = TokenType.COMMA;
-            case ';' -> tt = TokenType.SEMI;
-            case '+' -> tt = TokenType.PLUS;
-            case '=' -> tt = TokenType.EQUAL;
-            case '\0' -> tt = TokenType.EOF;
-        }
+        final var tokenType = switch (this.getCc()) {
+            case '{' -> TokenType.LSQIRLY;
+            case '}' -> TokenType.RSQIRLY;
+            case '(' -> TokenType.LPAREN;
+            case ')' -> TokenType.RPAREN;
+            case ',' -> TokenType.COMMA;
+            case ';' -> TokenType.SEMI;
+            case '+' -> TokenType.PLUS;
+            case '=' -> TokenType.EQUAL;
+            case '\0' -> TokenType.EOF;
+            default -> TokenType.ILLEGAL;
+        };
 
-        if (tt.equals(TokenType.EOF)) {
-            t = this.createToken(tt, "eof");
-        } else {
-            t = this.createToken(tt, String.valueOf(this.getCc()));
-        }
+        final var token = switch (tokenType) {
+            case EOF -> this.createToken(tokenType, "eof");
+            default -> this.createToken(tokenType, String.valueOf(this.getCc()));
+        };
+
+        if (Character.isLetter(this.getCc())) {
+            var ident = this.indent();
 
-        if (this.isLetter(this.getCc())) {
-            String ident = this.indent();
-            Token keyword = this.keywordMap.getOrDefault(ident, null);
-            if (keyword != null) {
-                return keyword;
-            } else {
-                return createToken(TokenType.IDENT, ident);
-            }
-        } else if (this.isNumber(this.getCc())) {
+            return Optional
```

Comment:
Why is this necessary instead of simply returning the found identifier or `ILLEGAL`?